### ISSUE TITLE
Allow custom failure when timeout exceeded

### ISSUE THREAD

**Issue by jhalterman** [jhalterman/failsafe#225]

**Comment by whiskeysierra** [comment]
Registering a `Fallback` that acts upon a `TimeoutExceededExcecption` doesn't work?

**Comment by jhalterman** [comment]
Yep that would be the current approach. I'm wondering if it's worth doing something more convenient though.

**Comment by whiskeysierra** [comment]
It's based on existing features and works exactly like any other fallback.
I like it.

**Comment by jhalterman** [comment]
Updated the OP to give some clearer code level examples of what this means.

**Comment by paulius-p** [comment]
Allowing something like this could also enhance `RetryPolicy` #202 ?
